client/src/app/components/client-gridview/client-gridview.component.html

```diff
@@ -1,28 +1,44 @@
 <div class="mat-elevation-z8">
     <table mat-table [dataSource]="dataSource" matSort>
   
```

Comment:
Although it's not my preferred way of modifying the table directly you made it work but there is still one problem here. When I modify one row of  First name one it is modified for some reason it is modifying the one that is under it also so I think you might have a problem with using two times the same ID or something like that. This also applies to IsActive and Lastname.